src/modules/hide_vodcasts/index.js

```diff
@@ -0,0 +1,46 @@
+const $ = require('jquery');
+const settings = require('../../settings');
+const watcher = require('../../watcher');
+
+class HideVodcastsModule {
+    constructor() {
+        settings.add({
+            id: 'hideVodcasts',
+            name: 'Hide Vodcasts',
+            defaultValue: false,
+            description: 'Hide Vodcasts in your following list'
+        });
+        settings.on('changed.hideVodcasts', value => value === true ? this.load() : this.unload());
+        watcher.on('load.following', () => this.load());
+    }
+
+    load() {
+        if (settings.get('hideVodcasts') === false) return;
+        $('.qa-stream-preview .is-watch-party').closest('.qa-stream-preview:visible').hide(); // hide vodcasts on page load
+        this.registerMutationObserver();
+    }
+
+    unload() {
+        if ( this.observer ) {
+            this.observer.disconnect();
+            $('.qa-stream-preview .is-watch-party').closest('.qa-stream-preview:hidden').show();
+        }
+    }
+
+    registerMutationObserver() {
```

Comment:
in general we do not put mutation observers anywhere but the watcher.js file.. this is because if someone wants to build something relying on the same/similar mutations we dont need multiple of the observer